integration-tests/test_proxy.py

```diff
@@ -53,14 +55,64 @@ def test_manual_https(preserve_config):
     # verify that our certificate was loaded by traefik
     assert server_cert == file_cert
 
-    for i in range(5):
+    for i in range(10):
         time.sleep(i)
         # verify that we can still connect to the hub
-        r = requests.get("https://127.0.0.1/hub/api", verify=False)
-        if r.status_code == 200:
-            break;
-
-    r.raise_for_status()
+        try:
+            req = HTTPRequest(
+                "https://127.0.0.1/hub/api", method="GET", validate_cert=False
+            )
+            resp = HTTPClient().fetch(req)
+            break
+        except Exception as e:
+            pass
+    assert resp.code == 200
 
     # cleanup
     shutil.rmtree(ssl_dir)
+
+
+def test_extra_traefik_config():
+    extra_config_dir = os.path.join(CONFIG_DIR, "traefik_config.d")
+    os.makedirs(extra_config_dir, exist_ok=True)
+
+    extra_config = {
+        "entryPoints": {"no_auth_api": {"address": "127.0.0.1:9999"}},
+        "api": {"dashboard": True, "entrypoint": "no_auth_api"},
+    }
+
+    success = False
+    for i in range(5):
+        time.sleep(i)
+        try:
+            with pytest.raises(HTTPClientError, match="HTTP 401: Unauthorized"):
+                # The default dashboard entrypoint requires authentication, so it should fail
+                req = HTTPRequest("http://127.0.0.1:8099/dashboard/", method="GET")
+                HTTPClient().fetch(req)
```

Comment:
This shouldn't return 200 because I'm not passing the credentials through the request, but it's expecting a 401:
 ```
with pytest.raises(HTTPClientError, match="HTTP 401: Unauthorized")
```
The boolean is to check that we exited the loop because we got the exception.

I could pass the credentials and verify with a 200 code instead? What do you think?

Reply:
Aaaah, I missed the raises! This looks great, sorry about that!